modules/generator/instance.go

```diff
@@ -113,60 +116,92 @@ func (i *instance) watchOverrides() {
 	}
 }
 
-func (i *instance) updateProcessors(desiredProcessors map[string]struct{}) error {
+func (i *instance) updateProcessors() error {
+	desiredProcessors := i.overrides.MetricsGeneratorProcessors(i.instanceID)
+	desiredCfg := i.cfg.Processor.copyWithOverrides(i.overrides, i.instanceID)
+
 	i.processorsMtx.RLock()
-	toAdd, toRemove := i.diffProcessors(desiredProcessors)
+	toAdd, toRemove, toReplace, err := i.diffProcessors(desiredProcessors, desiredCfg)
 	i.processorsMtx.RUnlock()
 
-	if len(toAdd) == 0 && len(toRemove) == 0 {
+	if err != nil {
+		return err
+	}
+	if len(toAdd) == 0 && len(toRemove) == 0 && len(toReplace) == 0 {
 		return nil
 	}
 
 	i.processorsMtx.Lock()
 	defer i.processorsMtx.Unlock()
 
 	for _, processorName := range toAdd {
-		err := i.addProcessor(processorName)
+		err := i.addProcessor(processorName, desiredCfg)
 		if err != nil {
 			return err
 		}
 	}
 	for _, processorName := range toRemove {
 		i.removeProcessor(processorName)
 	}
+	for _, processorName := range toReplace {
+		i.removeProcessor(processorName)
+
+		err := i.addProcessor(processorName, desiredCfg)
+		if err != nil {
+			return err
+		}
+	}
 
 	i.updateProcessorMetrics()
 
 	return nil
 }
 
-// diffProcessors compares the existings processors with desiredProcessors. Must be called under a
-// read lock.
-func (i *instance) diffProcessors(desiredProcessors map[string]struct{}) (toAdd []string, toRemove []string) {
+// diffProcessors compares the existing processors with the desired processors and config.
+// Must be called under a read lock.
+func (i *instance) diffProcessors(desiredProcessors map[string]struct{}, desiredCfg ProcessorConfig) (toAdd, toRemove, toReplace []string, err error) {
 	for processorName := range desiredProcessors {
 		if _, ok := i.processors[processorName]; !ok {
 			toAdd = append(toAdd, processorName)
 		}
 	}
-	for processorName := range i.processors {
+	for processorName, processor := range i.processors {
 		if _, ok := desiredProcessors[processorName]; !ok {
 			toRemove = append(toRemove, processorName)
+			continue
+		}
+		switch processorName {
```

Comment:
instead of adding `Config() interface{}` to the processor interface i'd consider just doing a type switch here on `processor` and accessing the processors config directly once you know the type.